examples/bevy_ggrs/src/main.rs

```diff
@@ -91,14 +88,10 @@ fn start_matchbox_socket(mut commands: Commands, args: Res<Args>) {
 
     let room_url = format!("{}/{}", &args.matchbox, room_id);
     info!("connecting to matchbox server: {:?}", room_url);
-    let (socket, message_loop) = WebRtcSocket::new_ggrs(room_url);
-
-    // The message loop needs to be awaited, or nothing will happen.
-    // We do this here using bevy's task system.
-    let task_pool = IoTaskPool::get();
-    task_pool.spawn(message_loop).detach();
 
-    commands.insert_resource(SocketResource(Some(socket)));
+    commands.add(OpenSocket(
+        WebRtcSocketBuilder::new(room_url).add_ggrs_channel(),
+    ));
```

Comment:
We could consider adding an extension trait if we want to have:

```rust
commands.matchbox_connect(url); // opens default reliable and unreliable channels (when we have that)
```

and/or:

```rust
commands.matchbox_connect(url).add_ggrs_channel();
```

Though the last one would require a `&mut self` builder, or we'd have to wrap the builder api as well.

Alternatively

```rust
commands.matchbox_connect(url).with(|b| b.add_ggrs_channel());
```

Though that's more clunky.

Reply:
Have added a couple extension traits called `open_socket` and `close_socket` though perhaps something with matchbox in the name would be more appropriate...

Right now `open_socket` takes a `WebRtcSocket` builder as it's only argument, not entirely sure how we would go about adding trailing functions without forcing to user to `.build()` which feels to me like a repetition of the builder itself

Reply:
Gotcha